packages/medusa/src/services/cart.ts

```diff
@@ -659,7 +665,164 @@ class CartService extends TransactionBaseService {
             { cart_id: cartId, has_shipping: true },
             { has_shipping: false }
           )
-          .catch(() => void 0)
+          .catch((err: Error) => {
+            // We only want to catch the errors related to not found items since we don't care if there is not item to update
+            if (!err.message?.endsWith("was not found")) {
```

Comment:
thought: This does not seem super scalable. Could we check the error type instead?

Reply:
yes you are right 👍

Reply:
A thought here - does it make sense to have the update function throw a not found error? It is not something we generally do across other services.